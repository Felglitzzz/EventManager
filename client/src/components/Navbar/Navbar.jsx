import React from 'react';

import { Link } from 'react-router-dom';

import SignInPageModal from '../UserAccess/SignInPageModal';
import SignUpPageModal from '../UserAccess/SignUpPageModal';

/**
 * @description - Stateless component for rendering navbar
 *
 * @returns {jsx} jsx - renders Navbar component
 */
const Navbar = () => (
  <div>
    <nav className="navbar navbar-expand-lg transparent navbar-dark bg-orange">
      <Link className="navbar-brand font-weight-bold text-white"
        to="/"
      ><i className="text-center fab fa-quinscape fa-2x mr-2" />
          EVENTERIA
      </Link>
      <button
        aria-controls="navbarSupportedContent"
        aria-expanded="false"
        aria-label="Toggle navigation"
        className="navbar-toggler"
        data-target="#navbarSupportedContent"
        data-toggle="collapse"
        type="button"
      >
        <span className="navbar-toggler-icon" />
      </button>
      <div className="collapse navbar-collapse"
        id="navbarSupportedContent"
      >
        <form className="form-inline ml-auto mx-2">
          <button
            className="btn-resp-height btn btn-outline-orange-1 rounded-0 px-3 ml-2 border"
            data-target="#signinModal"
            data-toggle="modal"
            type="button"
          >
            <i aria-hidden="true"
              className="fa fa-user"
            >
              {' '}
                Sign In
            </i>
          </button>
        </form>
        <form className="form-inline">
          <button
            className="btn-resp-height btn btn-outline-orange-1 rounded-0 px-3 ml-2 border"
            data-target="#signupModal"
            data-toggle="modal"
            type="button"
          >
            <i aria-hidden="true"
              className="fa fa-user-plus"
            >
              {' '}
                Sign Up
            </i>
          </button>
        </form>
      </div>
    </nav>

    {/* Modal for sign in */}
    <SignInPageModal />

    {/* Modal for sign up */}
    <SignUpPageModal />
  </div>
);

export default Navbar;
