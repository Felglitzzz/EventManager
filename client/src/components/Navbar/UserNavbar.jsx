import React from 'react';
import { connect } from 'react-redux';
import lodash from 'lodash';
import { logOutUser } from '../../actions/userAccessActions';
import getUserInfo from '../../utils/getUserFromToken';

/**
 * @description - Class component for rendering authenticated navbar
 *
 * @class UserNavbar
 *
 * @returns {jsx} jsx - renders UserNavbar component
 */
class UserNavbar extends React.Component {
  /**
  * @description - renders dynamic navbar for authenticated users
  *
  * @returns { jsx } jsx - renders authenticated navbar component
  *
  * @memberof UserNavbar
  */
  render() {
    const { username } = getUserInfo();
    return (
      <div className="no-padding-hr container mb-3 hoverable">
        <nav className="navbar navbar-expand-lg d-flex">
          <p className="font-weight-bold montfont text-orange ml-auto myAuto">
            Hello, {lodash.capitalize(username)}
          </p>
        </nav>
      </div>
    );
  }
}

/**
 * @param {func} dispatch
 * @returns {object} dipatched action
 */
const mapDispatchToProps = dispatch => ({
  logOutUser: () => dispatch(logOutUser())
});

export default connect(mapDispatchToProps)(UserNavbar);
