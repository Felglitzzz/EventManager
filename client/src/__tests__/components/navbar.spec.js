import React from 'react';
import { shallow } from 'enzyme';
import Navbar from '../../components/Navbar/Navbar';
import { UserNavbar, mapDispatchToProps } from '../../components/Navbar/UserNavbar';
import mockLocalStorage from '../__mocks__/mockLocalStorage';

window.localStorage = mockLocalStorage;

describe('Navbar Page', () => {
  it('should render the navbar page correctly', () => {
    const wrapper = shallow(<Navbar />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe('UserNavbar component', () => {
  it('should render the usernavbar component', () => {
    const wrapper = shallow(<UserNavbar
      logOutUser = {() => { Promise.resolve(); }}
    />);
    expect(wrapper.length).toEqual(1);
  });

  it('should test mapDispatchToProps method', () => {
    const dispatch = jest.fn();
    expect(mapDispatchToProps(dispatch).logOutUser).toBeTruthy();
  });
});
