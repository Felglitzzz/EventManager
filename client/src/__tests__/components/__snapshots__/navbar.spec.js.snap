// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar Page should render the navbar page correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Navbar />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <nav
          className="navbar navbar-expand-lg transparent navbar-dark bg-orange"
        >
          <Link
            className="navbar-brand font-weight-bold text-white"
            replace={false}
            to="/"
          >
            <i
              className="text-center fab fa-quinscape fa-2x mr-2"
            />
            EVENTERIA
          </Link>
          <button
            aria-controls="navbarSupportedContent"
            aria-expanded="false"
            aria-label="Toggle navigation"
            className="navbar-toggler"
            data-target="#navbarSupportedContent"
            data-toggle="collapse"
            type="button"
          >
            <span
              className="navbar-toggler-icon"
            />
          </button>
          <div
            className="collapse navbar-collapse"
            id="navbarSupportedContent"
          >
            <form
              className="form-inline ml-auto mx-2"
            >
              <button
                className="btn-resp-height btn btn-outline-orange-1 rounded-0 px-3 ml-2 border"
                data-target="#signinModal"
                data-toggle="modal"
                type="button"
              >
                <i
                  aria-hidden="true"
                  className="fa fa-user"
                >
                   
                  Sign In
                </i>
              </button>
            </form>
            <form
              className="form-inline"
            >
              <button
                className="btn-resp-height btn btn-outline-orange-1 rounded-0 px-3 ml-2 border"
                data-target="#signupModal"
                data-toggle="modal"
                type="button"
              >
                <i
                  aria-hidden="true"
                  className="fa fa-user-plus"
                >
                   
                  Sign Up
                </i>
              </button>
            </form>
          </div>
        </nav>,
        <Connect(SignInPageModal) />,
        <Connect(SignUpPageModal) />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Link
              className="navbar-brand font-weight-bold text-white"
              replace={false}
              to="/"
            >
              <i
                className="text-center fab fa-quinscape fa-2x mr-2"
              />
              EVENTERIA
            </Link>,
            <button
              aria-controls="navbarSupportedContent"
              aria-expanded="false"
              aria-label="Toggle navigation"
              className="navbar-toggler"
              data-target="#navbarSupportedContent"
              data-toggle="collapse"
              type="button"
            >
              <span
                className="navbar-toggler-icon"
              />
            </button>,
            <div
              className="collapse navbar-collapse"
              id="navbarSupportedContent"
            >
              <form
                className="form-inline ml-auto mx-2"
              >
                <button
                  className="btn-resp-height btn btn-outline-orange-1 rounded-0 px-3 ml-2 border"
                  data-target="#signinModal"
                  data-toggle="modal"
                  type="button"
                >
                  <i
                    aria-hidden="true"
                    className="fa fa-user"
                  >
                     
                    Sign In
                  </i>
                </button>
              </form>
              <form
                className="form-inline"
              >
                <button
                  className="btn-resp-height btn btn-outline-orange-1 rounded-0 px-3 ml-2 border"
                  data-target="#signupModal"
                  data-toggle="modal"
                  type="button"
                >
                  <i
                    aria-hidden="true"
                    className="fa fa-user-plus"
                  >
                     
                    Sign Up
                  </i>
                </button>
              </form>
            </div>,
          ],
          "className": "navbar navbar-expand-lg transparent navbar-dark bg-orange",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <i
                  className="text-center fab fa-quinscape fa-2x mr-2"
                />,
                "EVENTERIA",
              ],
              "className": "navbar-brand font-weight-bold text-white",
              "replace": false,
              "to": "/",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "text-center fab fa-quinscape fa-2x mr-2",
                },
                "ref": null,
                "rendered": null,
                "type": "i",
              },
              "EVENTERIA",
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "aria-controls": "navbarSupportedContent",
              "aria-expanded": "false",
              "aria-label": "Toggle navigation",
              "children": <span
                className="navbar-toggler-icon"
              />,
              "className": "navbar-toggler",
              "data-target": "#navbarSupportedContent",
              "data-toggle": "collapse",
              "type": "button",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "navbar-toggler-icon",
              },
              "ref": null,
              "rendered": null,
              "type": "span",
            },
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <form
                  className="form-inline ml-auto mx-2"
                >
                  <button
                    className="btn-resp-height btn btn-outline-orange-1 rounded-0 px-3 ml-2 border"
                    data-target="#signinModal"
                    data-toggle="modal"
                    type="button"
                  >
                    <i
                      aria-hidden="true"
                      className="fa fa-user"
                    >
                       
                      Sign In
                    </i>
                  </button>
                </form>,
                <form
                  className="form-inline"
                >
                  <button
                    className="btn-resp-height btn btn-outline-orange-1 rounded-0 px-3 ml-2 border"
                    data-target="#signupModal"
                    data-toggle="modal"
                    type="button"
                  >
                    <i
                      aria-hidden="true"
                      className="fa fa-user-plus"
                    >
                       
                      Sign Up
                    </i>
                  </button>
                </form>,
              ],
              "className": "collapse navbar-collapse",
              "id": "navbarSupportedContent",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <button
                    className="btn-resp-height btn btn-outline-orange-1 rounded-0 px-3 ml-2 border"
                    data-target="#signinModal"
                    data-toggle="modal"
                    type="button"
                  >
                    <i
                      aria-hidden="true"
                      className="fa fa-user"
                    >
                       
                      Sign In
                    </i>
                  </button>,
                  "className": "form-inline ml-auto mx-2",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <i
                      aria-hidden="true"
                      className="fa fa-user"
                    >
                       
                      Sign In
                    </i>,
                    "className": "btn-resp-height btn btn-outline-orange-1 rounded-0 px-3 ml-2 border",
                    "data-target": "#signinModal",
                    "data-toggle": "modal",
                    "type": "button",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "aria-hidden": "true",
                      "children": Array [
                        " ",
                        "Sign In",
                      ],
                      "className": "fa fa-user",
                    },
                    "ref": null,
                    "rendered": Array [
                      " ",
                      "Sign In",
                    ],
                    "type": "i",
                  },
                  "type": "button",
                },
                "type": "form",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <button
                    className="btn-resp-height btn btn-outline-orange-1 rounded-0 px-3 ml-2 border"
                    data-target="#signupModal"
                    data-toggle="modal"
                    type="button"
                  >
                    <i
                      aria-hidden="true"
                      className="fa fa-user-plus"
                    >
                       
                      Sign Up
                    </i>
                  </button>,
                  "className": "form-inline",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <i
                      aria-hidden="true"
                      className="fa fa-user-plus"
                    >
                       
                      Sign Up
                    </i>,
                    "className": "btn-resp-height btn btn-outline-orange-1 rounded-0 px-3 ml-2 border",
                    "data-target": "#signupModal",
                    "data-toggle": "modal",
                    "type": "button",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "aria-hidden": "true",
                      "children": Array [
                        " ",
                        "Sign Up",
                      ],
                      "className": "fa fa-user-plus",
                    },
                    "ref": null,
                    "rendered": Array [
                      " ",
                      "Sign Up",
                    ],
                    "type": "i",
                  },
                  "type": "button",
                },
                "type": "form",
              },
            ],
            "type": "div",
          },
        ],
        "type": "nav",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <nav
            className="navbar navbar-expand-lg transparent navbar-dark bg-orange"
          >
            <Link
              className="navbar-brand font-weight-bold text-white"
              replace={false}
              to="/"
            >
              <i
                className="text-center fab fa-quinscape fa-2x mr-2"
              />
              EVENTERIA
            </Link>
            <button
              aria-controls="navbarSupportedContent"
              aria-expanded="false"
              aria-label="Toggle navigation"
              className="navbar-toggler"
              data-target="#navbarSupportedContent"
              data-toggle="collapse"
              type="button"
            >
              <span
                className="navbar-toggler-icon"
              />
            </button>
            <div
              className="collapse navbar-collapse"
              id="navbarSupportedContent"
            >
              <form
                className="form-inline ml-auto mx-2"
              >
                <button
                  className="btn-resp-height btn btn-outline-orange-1 rounded-0 px-3 ml-2 border"
                  data-target="#signinModal"
                  data-toggle="modal"
                  type="button"
                >
                  <i
                    aria-hidden="true"
                    className="fa fa-user"
                  >
                     
                    Sign In
                  </i>
                </button>
              </form>
              <form
                className="form-inline"
              >
                <button
                  className="btn-resp-height btn btn-outline-orange-1 rounded-0 px-3 ml-2 border"
                  data-target="#signupModal"
                  data-toggle="modal"
                  type="button"
                >
                  <i
                    aria-hidden="true"
                    className="fa fa-user-plus"
                  >
                     
                    Sign Up
                  </i>
                </button>
              </form>
            </div>
          </nav>,
          <Connect(SignInPageModal) />,
          <Connect(SignUpPageModal) />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Link
                className="navbar-brand font-weight-bold text-white"
                replace={false}
                to="/"
              >
                <i
                  className="text-center fab fa-quinscape fa-2x mr-2"
                />
                EVENTERIA
              </Link>,
              <button
                aria-controls="navbarSupportedContent"
                aria-expanded="false"
                aria-label="Toggle navigation"
                className="navbar-toggler"
                data-target="#navbarSupportedContent"
                data-toggle="collapse"
                type="button"
              >
                <span
                  className="navbar-toggler-icon"
                />
              </button>,
              <div
                className="collapse navbar-collapse"
                id="navbarSupportedContent"
              >
                <form
                  className="form-inline ml-auto mx-2"
                >
                  <button
                    className="btn-resp-height btn btn-outline-orange-1 rounded-0 px-3 ml-2 border"
                    data-target="#signinModal"
                    data-toggle="modal"
                    type="button"
                  >
                    <i
                      aria-hidden="true"
                      className="fa fa-user"
                    >
                       
                      Sign In
                    </i>
                  </button>
                </form>
                <form
                  className="form-inline"
                >
                  <button
                    className="btn-resp-height btn btn-outline-orange-1 rounded-0 px-3 ml-2 border"
                    data-target="#signupModal"
                    data-toggle="modal"
                    type="button"
                  >
                    <i
                      aria-hidden="true"
                      className="fa fa-user-plus"
                    >
                       
                      Sign Up
                    </i>
                  </button>
                </form>
              </div>,
            ],
            "className": "navbar navbar-expand-lg transparent navbar-dark bg-orange",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <i
                    className="text-center fab fa-quinscape fa-2x mr-2"
                  />,
                  "EVENTERIA",
                ],
                "className": "navbar-brand font-weight-bold text-white",
                "replace": false,
                "to": "/",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "text-center fab fa-quinscape fa-2x mr-2",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "i",
                },
                "EVENTERIA",
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "aria-controls": "navbarSupportedContent",
                "aria-expanded": "false",
                "aria-label": "Toggle navigation",
                "children": <span
                  className="navbar-toggler-icon"
                />,
                "className": "navbar-toggler",
                "data-target": "#navbarSupportedContent",
                "data-toggle": "collapse",
                "type": "button",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "navbar-toggler-icon",
                },
                "ref": null,
                "rendered": null,
                "type": "span",
              },
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <form
                    className="form-inline ml-auto mx-2"
                  >
                    <button
                      className="btn-resp-height btn btn-outline-orange-1 rounded-0 px-3 ml-2 border"
                      data-target="#signinModal"
                      data-toggle="modal"
                      type="button"
                    >
                      <i
                        aria-hidden="true"
                        className="fa fa-user"
                      >
                         
                        Sign In
                      </i>
                    </button>
                  </form>,
                  <form
                    className="form-inline"
                  >
                    <button
                      className="btn-resp-height btn btn-outline-orange-1 rounded-0 px-3 ml-2 border"
                      data-target="#signupModal"
                      data-toggle="modal"
                      type="button"
                    >
                      <i
                        aria-hidden="true"
                        className="fa fa-user-plus"
                      >
                         
                        Sign Up
                      </i>
                    </button>
                  </form>,
                ],
                "className": "collapse navbar-collapse",
                "id": "navbarSupportedContent",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <button
                      className="btn-resp-height btn btn-outline-orange-1 rounded-0 px-3 ml-2 border"
                      data-target="#signinModal"
                      data-toggle="modal"
                      type="button"
                    >
                      <i
                        aria-hidden="true"
                        className="fa fa-user"
                      >
                         
                        Sign In
                      </i>
                    </button>,
                    "className": "form-inline ml-auto mx-2",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <i
                        aria-hidden="true"
                        className="fa fa-user"
                      >
                         
                        Sign In
                      </i>,
                      "className": "btn-resp-height btn btn-outline-orange-1 rounded-0 px-3 ml-2 border",
                      "data-target": "#signinModal",
                      "data-toggle": "modal",
                      "type": "button",
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "aria-hidden": "true",
                        "children": Array [
                          " ",
                          "Sign In",
                        ],
                        "className": "fa fa-user",
                      },
                      "ref": null,
                      "rendered": Array [
                        " ",
                        "Sign In",
                      ],
                      "type": "i",
                    },
                    "type": "button",
                  },
                  "type": "form",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <button
                      className="btn-resp-height btn btn-outline-orange-1 rounded-0 px-3 ml-2 border"
                      data-target="#signupModal"
                      data-toggle="modal"
                      type="button"
                    >
                      <i
                        aria-hidden="true"
                        className="fa fa-user-plus"
                      >
                         
                        Sign Up
                      </i>
                    </button>,
                    "className": "form-inline",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <i
                        aria-hidden="true"
                        className="fa fa-user-plus"
                      >
                         
                        Sign Up
                      </i>,
                      "className": "btn-resp-height btn btn-outline-orange-1 rounded-0 px-3 ml-2 border",
                      "data-target": "#signupModal",
                      "data-toggle": "modal",
                      "type": "button",
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "aria-hidden": "true",
                        "children": Array [
                          " ",
                          "Sign Up",
                        ],
                        "className": "fa fa-user-plus",
                      },
                      "ref": null,
                      "rendered": Array [
                        " ",
                        "Sign Up",
                      ],
                      "type": "i",
                    },
                    "type": "button",
                  },
                  "type": "form",
                },
              ],
              "type": "div",
            },
          ],
          "type": "nav",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
